programs/survival/cam.sc

```diff
@@ -164,3 +170,45 @@ __survival_defaults(player) ->
    print(format('rb Cannot find a safe spot to land within 32 blocks.'));
    false;
 );
+
+__restrict_flight(player) -> 
+(
+   if(global_max_flight_range > 0, 
+      config = __get_player_stored_takeoff_params(player~'name');
+      if(!config,  // gotta restrict it somehow if no config was saved
+         start_pos = player~'pos';
+         start_dim = player~'dimension',
+
+         start_pos = config:'pos'; 
+         start_dim = config:'dimension'
+      );
+      
+      __restrict_flight_tick(player, start_pos, start_dim)
+   )
+);
+
+__restrict_flight_tick(player, start_pos, start_dim) -> 
+(
+   if(player~'gamemode' != 'spectator', exit());
+
+   pp = player~'pos';
+   if(_euclidean(pp, start_pos)>global_max_flight_range,
```

Comment:
U typically do `_euclidean_sq(pp, start_pos) > global_max_flight_range * global_max_flight_range` cos that way u avoid sqrt

Reply:
Yeah, it's just that I didn't want to store the squared max radius globally, and it's not that we arecalculating thousands of square roots every tick. I can remove it if it bothers you.

Reply:
Well, not anymore, because I use the same distance value for both absolute value and euclidian distance.